pkgs/by-name/bu/buildstream/package.nix

```diff
@@ -0,0 +1,67 @@
+{
+  python3Packages,
+  fetchPypi,
+  lzip,
+  patch,
+  bubblewrap,
+  fuse3,
+  lib,
+  buildbox,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "buildstream";
+  version = "2.4.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-crjsC66S4L42dejUQQ+i9j4+W68JF9Dn6kezaN3EZF8=";
+  };
+
+  build-system = with python3Packages; [ setuptools ];
+
+  dependencies = with python3Packages; [
+    click
+    dulwich
+    grpcio
+    jinja2
+    markupsafe
+    packaging
+    pluginbase
+    protobuf
+    psutil
+    pyroaring
+    requests
+    ruamel-yaml
+    ruamel-yaml-clib
+    tomlkit
+    ujson
+  ];
+
+  propagatedBuildInputs = [
+    buildbox
+    fuse3
+    lzip
+    patch
+    python3Packages.cython
+  ];
+
+  nativeBuildInputs = with python3Packages; [
+    bubblewrap
+    pdm-pep517
+    setuptools-scm
```

Comment:
Should go in `build-system`.

Reply:
I was meaning only `setuptools-scm`, but yes, it looks like `pdm-pep517` too.

Reply:
I guess only `setuptools-scm`. One good way to learn is to search in `nixpkgs` for `setuptools-scm` and see where it is mostly used.